server/internal/server/completions_test.go

```diff
@@ -46,231 +46,141 @@ func TestCreateChatCompletion(t *testing.T) {
 			},
 		},
 		&fakeRewriter{},
-		&fakeTaskSender{
-			resp: &http.Response{
-				StatusCode: http.StatusOK,
-				Body:       io.NopCloser(bytes.NewReader([]byte{})),
-			}},
+		capturingTaskSender,
 		logger,
 	)
 	srv.enableAuth = true
 
 	tcs := []struct {
-		name string
-		req  *v1.CreateChatCompletionRequest
-		code int
+		name    string
+		req     *v1.CreateChatCompletionRequest
+		expCode int
+		assert  func(t *testing.T, req *v1.CreateChatCompletionRequest)
 	}{
 		{
-			name: "success",
-			req: &v1.CreateChatCompletionRequest{
-				Model: modelID,
-				Messages: []*v1.CreateChatCompletionRequest_Message{
-					{
-						Role: "user",
-						Content: []*v1.CreateChatCompletionRequest_Message_Content{
-							{
-								Type: "text",
-								Text: "test",
-							},
-						},
-					},
-				},
-			},
-			code: http.StatusOK,
+			name:    "success",
+			req:     CreateChatCompletionRequestFixture(WithModel(modelID)),
```

Comment:
I have introduced a fixture for the common request defaults. The old test setup was fine but this makes it a bit more readable IMHO as the changes are more visible.